core/trino-main/src/main/java/io/trino/sql/planner/optimizations/UnaliasSymbolReferences.java

```diff
@@ -1303,4 +1331,37 @@ public PlanNode getRoot()
             return mappings;
         }
     }
+
+    private static class DynamicFilterVisitor
+            extends SimplePlanRewriter<Void>
+    {
+        private final Metadata metadata;
+        private final Map<DynamicFilterId, DynamicFilterId> dynamicFilterIdMap;
+
+        private DynamicFilterVisitor(Metadata metadata, Map<DynamicFilterId, DynamicFilterId> dynamicFilterIdMap)
+        {
+            this.metadata = requireNonNull(metadata, "metadata is null");
+            this.dynamicFilterIdMap = requireNonNull(dynamicFilterIdMap, "dynamicFilterIdMap is null");
+        }
+
+        @Override
+        public PlanNode visitFilter(FilterNode node, RewriteContext<Void> context)
+        {
+            return new FilterNode(
+                    node.getId(),
+                    context.rewrite(node.getSource()),
+                    updateDynamicFilterIds(dynamicFilterIdMap, node.getPredicate()));
```

Comment:
you should return same instance if no filters are rewritten

Reply:
Ok, I changed it to return the same instance if predicate and source were not rewritten.